projects/plugins/jetpack/extensions/blocks/donations/edit.js

```diff
@@ -1,102 +1,187 @@
+	const [ products, setProducts ] = useState( defaultProducts );
+	const [ activeTab, setActiveTab ] = useState( DEFAULT_TAB );
 
-	const post = useSelect( select => select( 'core/editor' ).getCurrentPost(), [] );
-	useEffect( () => {
-		setAttributes( { fallbackLinkUrl: post.link } );
-	}, [ post.link, setAttributes ] );
+	const fallbackLinkUrl = useSelect( select => select( 'core/editor' ).getCurrentPost(), [] ).link;
+	const postId = useSelect( select => select( 'core/editor' ).getCurrentPostId(), [] );
 
-	const apiError = message => {
-		setLoadingError( message );
+	const resetActiveTab = ( controlTab, value ) => {
+		if ( value === false && activeTab === controlTab ) {
+			setActiveTab( DEFAULT_TAB );
+		}
 	};
 
 	const filterProducts = productList =>
 		productList.reduce( ( filteredProducts, { id, currency: productCurrency, type, interval } ) => {
 			if ( productCurrency === currency && type === 'donation' ) {
-				filteredProducts[ interval ] = id;
+				const mappedInterval = intervalMap[ interval ];
+				if ( mappedInterval ) {
+					filteredProducts[ mappedInterval ] = id;
+				}
 			}
 			return filteredProducts;
 		}, {} );
 
 	const hasRequiredProducts = productIdsPerInterval => {
 		const intervals = Object.keys( productIdsPerInterval );
-
-		return (
-			intervals.includes( 'one-time' ) &&
-			intervals.includes( '1 month' ) &&
-			intervals.includes( '1 year' )
-		);
+		return Object.keys( defaultProducts ).every( required => intervals.includes( required ) );
 	};
 
 	const mapStatusToState = result => {
-		if ( ( ! result && typeof result !== 'object' ) || result.errors ) {
-			setLoadingError( __( 'Could not load data from WordPress.com.', 'jetpack' ) );
-			return;
-		}
 		setShouldUpgrade( result.should_upgrade_to_access_memberships );
 		setStripeConnectUrl( result.connect_url );
 
+		if ( result.should_upgrade_to_access_memberships || result.connect_url ) {
+			setProducts( defaultProducts );
+			return;
+		}
+
 		const filteredProducts = filterProducts( result.products );
 
 		if ( hasRequiredProducts( filteredProducts ) ) {
 			setProducts( filteredProducts );
 			return;
 		}
 
-		// Set fake products when plan should be upgraded or there is no connection to Stripe so users can still try the
-		// block in the editor.
-		if ( result.should_upgrade_to_access_memberships || result.connect_url ) {
-			setProducts( {
-				'one-time': -1,
-				'1 month': -1,
-				'1 year': -1,
-			} );
-			return;
-		}
-
 		// Only create products if we have the correct plan and stripe connection.
-		fetchDefaultProducts( currency ).then( defaultProducts => {
-			return setProducts( filterProducts( defaultProducts ) );
-		}, apiError );
+		fetchDefaultProducts( currency ).then( defaultFetchedProducts => {
+			return setProducts( filterProducts( defaultFetchedProducts ) );
+		}, setLoadingError );
 	};
 
 	useEffect( () => {
-		const updateData = () => fetchStatus( 'donation' ).then( mapStatusToState, apiError );
-		updateData();
```

Comment:
We were storing a function just to call it immediately after. Changed for the invocation of the function contents.